app/controllers/trainees/apply_applications/course_details_controller.rb

```diff
@@ -36,11 +36,15 @@ def set_course
       end
 
       def subject_specialism_path
-        return edit_trainee_language_specialisms_path(trainee) if specialism_type == :language
+        return edit_trainee_language_specialisms_path(trainee) if language_specialism?
 
         edit_trainee_subject_specialism_path(trainee, 1)
       end
 
+      def language_specialism?
+        specialism_type == :language || specialism_type == :language_and_other
+      end
```

Comment:
would shifting `CalculateSubjectSpecialismType#language_specialism?` to public access help here?

Reply:
I did wonder about that but it kind of breaks the `ServicePattern` do one thing pattern.  I'll try to work out a way to share the logic though...

Reply:
Actually on closer inspection it looks like the two methods do different things. I'm going to rename one of them before I get any more confused.